src/libdrakvuf/linux-processes.c

```diff
@@ -267,3 +267,68 @@ bool linux_get_current_thread_id( drakvuf_t drakvuf, uint64_t vcpu_id, uint32_t
 
     return true;
 }
+
+vmi_pid_t linux_get_process_ppid( drakvuf_t drakvuf, addr_t process_base )
+{
+    uint32_t ppid = -1 ;
+    status_t ret = VMI_FAILURE ;
```

Comment:
I don't think there is need for this variable on the stack.

Reply:
I'm afraid is needed as is currently coded, anyway I can get rid of the initialization

Reply:
What I meant is that this variable is not used for anything else then checking the return value of the functions, so those could be checked directly instead of storing it in another variable first.

Reply:
I know what you mean but the way is coded you need the return value or you end up duplicating code.

Anyway, this function will be substantially changed. :)

Reply:
FYI, the "flow" is:

1- Try to get the pointer to ```real_parent``` of the current task_struct. If goes well go to step 4.
If fails go to step 2

2- Try to get the pointer to ```parent``` of the curren task_struct. If goes well go to step 4, if fails go to step 3

3- We were unable to get ```real_parent``` OR ```parent```, go to step 5

4- Arrived here we have an achieved parent, don't mind if it comes from ```real_parent``` or from ```parent```, we can get its TGID

5- Exit from function